src/server/api.rs

```diff
@@ -51,49 +53,67 @@ impl Project {
 }
 
 pub fn routes(state: Project) -> Router {
-    Router::new()
-        // general
-        .route("/about", get(about))
-        .route("/settings", get(settings_get).post(settings_update))
-        .route("/stats", get(stats))
-        .route("/session", get(session))
-        // books
-        .route("/book", get(book_search).post(book_add))
-        .route(
-            "/book/:id",
-            get(book_fetch).post(book_update).delete(book_delete),
-        )
-        .route("/book-id", post(book_generate_id))
-        .route("/book-fetch/:isbn", get(book_fetch_data))
-        // user
-        .route("/user", get(user_search).post(user_add))
-        .route(
-            "/user/:account",
-            get(user_fetch).post(user_update).delete(user_delete),
-        )
-        .route("/user-fetch/:account", get(user_fetch_data))
-        .route("/user-update-roles", post(user_update_roles))
-        // category
-        .route("/category", get(category_list).post(category_add))
-        .route(
-            "/category/:id",
-            post(category_update).delete(category_delete),
-        )
-        .route("/category-refs/:id", get(category_references))
-        // lending
-        .route("/lending/lend", post(lending_lend))
-        .route("/lending/return", post(lending_return))
-        .route("/lending/reserve", post(lending_reserve))
-        .route("/lending/release", post(lending_release))
-        .route("/overdues", get(lending_overdues))
-        // mail
-        .route("/notify", post(mail_notify))
-        // all routes require authorization
-        .route_layer(from_extractor_with_state::<Login, Auth>(state.auth.clone()))
+    let mut api = Router::new();
+
+    // general
+    route!(api, "/about", get(about));
```

Comment:
Könnte man das in ein Macro bekommen?

```rs
let router = gluer::generate!{
    prefix = "/api", // defaults to "/"
    routes = { // required
        "/about" = get(about),
        "/settings" = get(settings_get).post(settings_update),
        // ...
    },
    files = ["src/db", "src/server", "src/error.rs", "src/provider/dnb.rs"], // defaults to ["src"]
    output = "lib-view/src/lib/api.ts", //required
}
```

Reply:
Joa, wollte nur die "normale" router-definition nicht zu stark abändern, aber das sollte möglich sein. Setze mich daran.